crates/pessimistic-proof/src/batch_header.rs

```diff
@@ -0,0 +1,133 @@
+#![allow(clippy::too_many_arguments)]
+use std::{collections::BTreeMap, hash::Hash};
+
+use reth_primitives::U256;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use serde_with::serde_as;
+
+use crate::{
+    bridge_exit::{BridgeExit, NetworkId, TokenInfo},
+    imported_bridge_exit::ImportedBridgeExit,
+    local_balance_tree::LocalBalancePath,
+    local_exit_tree::hasher::Hasher,
+    nullifier_tree::NullifierPath,
+};
+
+/// Represents the data submitted by the CDKs to the AggLayer.
+///
+/// The bridge exits plus the imported bridge exits define
+/// the state transition, resp. the amount that goes out and the amount that comes in.
+///
+/// The bridge exits refer to the [`BridgeExit`]  emitted by
+/// the origin network of the [`BatchHeader`].
+///
+/// The imported bridge exits refer to the [`BridgeExit`] received and imported
+/// by the origin network of the [`BatchHeader].
+#[serde_as]
+#[derive(Default, Clone, Debug, Serialize, Deserialize)]
+pub struct BatchHeader<H>
+where
+    H: Hasher,
+    H::Digest: Eq + Hash + Copy + Serialize + DeserializeOwned,
+{
+    /// The origin network which emitted this BatchHeader.
+    /// TODO: we should clarify naming. We use origin to refer to the issuing network of a token. Could consider using "sending" or "local" to refer to the network that created a batch.
+    pub origin_network: NetworkId,
+
+    // /// The start slot height and end block height for the batch. Unsure if required
+    // pub start_block_height: u32,
+    // pub end_block_height u32,
+
+    // /// The state root produced by the current block
+    // TODO: Determine if this is necessary
+    // pub state_root: Option<Digest>,
+    /// The initial local exit root.
+    #[serde_as(as = "_")]
+    pub prev_local_exit_root: H::Digest,
+
+    /// The new local exit root
+    #[serde_as(as = "_")]
+    pub new_local_exit_root: H::Digest,
+
+    /// The set of bridge exits created in this batch
+    /// TODO: move out of the header and into a separate struct
+    pub bridge_exits: Vec<BridgeExit>,
+
+    /// The set of imported bridge exits claimed in this batch
+    /// TODO: move out of the header and into a separate struct
+    pub imported_bridge_exits: Vec<(ImportedBridgeExit, NullifierPath<H>)>,
+
+    /// A commitment to the set of imported bridge exits for which the origin network is the target.
+    #[serde_as(as = "Option<_>")]
+    pub imported_exits_root: Option<H::Digest>,
+
+    /// The set of imported local exit roots
+    // TODO: benchmark if BTreeMap is the best choice in terms of SP1 cycles
+    pub imported_local_exit_roots: BTreeMap<NetworkId, H::Digest>,
+
+    /// A map from token info to the token balance of the receiving network before any bridge event is processed,
```

Comment:
you meant the emitting/sending network of this `BatchHeader` right?

```suggestion
    /// A map from token info to the token balance of the origin network before any bridge event is processed,
```

Reply:
That's right, I meant "origin".